experimental/rsocket-src/RSocketServer.cpp

```diff
@@ -1,77 +1,44 @@
 // Copyright 2004-present Facebook. All Rights Reserved.
 
 #include "rsocket/RSocketServer.h"
+
 #include <folly/ExceptionWrapper.h>
-#include <folly/io/async/EventBase.h>
-#include <folly/io/async/EventBaseManager.h>
-#include "rsocket/OldNewBridge.h"
-#include "rsocket/RSocketErrors.h"
-#include "src/FrameTransport.h"
-#include "src/NullRequestHandler.h"
+
+#include "rsocket/RSocketConnectionHandler.h"
 
 using namespace reactivesocket;
 
 namespace rsocket {
 
-using OnSetupNewSocket = std::function<void(
-    std::shared_ptr<FrameTransport> frameTransport,
-    ConnectionSetupPayload setupPayload,
-    folly::Executor&)>;
-
-class RSocketConnectionHandler : public reactivesocket::ConnectionHandler {
+class RSocketServerConnectionHandler : public virtual RSocketConnectionHandler {
  public:
-  explicit RSocketConnectionHandler(OnSetupNewSocket onSetup)
-      : onSetup_(std::move(onSetup)) {}
-
-  ~RSocketConnectionHandler() {
-    LOG(INFO) << "RSocketServer => destroy the connection handler";
-  }
-
-  void setupNewSocket(
-      std::shared_ptr<FrameTransport> frameTransport,
-      ConnectionSetupPayload setupPayload) override {
-    // FIXME(alexanderm): Handler should be tied to specific executor
-    auto executor = folly::EventBaseManager::get()->getExistingEventBase();
-    onSetup_(std::move(frameTransport), std::move(setupPayload), *executor);
-  }
-
-  bool resumeSocket(
-      std::shared_ptr<FrameTransport> frameTransport,
-      ResumeParameters) override {
-    //      onSetup_(std::move(frameTransport), std::move(setupPayload));
-    return false;
+  RSocketServerConnectionHandler(RSocketServer* server, OnAccept onAccept) {
+    server_ = server;
+    onAccept_ = onAccept;
   }
 
-  void connectionError(
-      std::shared_ptr<FrameTransport>,
-      folly::exception_wrapper ex) override {
-    LOG(WARNING) << "Connection failed before first frame: " << ex.what();
+  std::shared_ptr<RSocketRequestHandler> getHandler(
+      std::shared_ptr<ConnectionSetupRequest> request) override {
+    return onAccept_(std::move(request));
   }
 
- private:
-  OnSetupNewSocket onSetup_;
-};
-
-class RSocketHandlerBridge : public reactivesocket::DefaultRequestHandler {
- public:
-  RSocketHandlerBridge(std::shared_ptr<RSocketRequestHandler> handler)
-      : handler_(std::move(handler)){};
-
-  void handleRequestStream(
-      Payload request,
-      StreamId streamId,
-      const std::shared_ptr<Subscriber<Payload>>&
-          subscriber) noexcept override {
-    auto flowable =
-        handler_->handleRequestStream(std::move(request), std::move(streamId));
-    // bridge from the existing eager RequestHandler and old Subscriber type
-    // to the lazy Flowable and new Subscriber type
-    flowable->subscribe(yarpl::Reference<yarpl::flowable::Subscriber<Payload>>(
-        new NewToOldSubscriber(std::move(subscriber))));
+  void manageSocket(
+      std::shared_ptr<ConnectionSetupRequest> request,
```

Comment:
got the answer:
" I added it because I need the context in order to associate the socket with the right Session
otherwise I have a socket but no way to know what Session it goes with
Yeah it's not used in the sample code"

makes sense to me.